packages/react-components/react-overflow/src/useOverflowContainer.ts

```diff
@@ -65,57 +69,9 @@ export const useOverflowContainer = <TElement extends HTMLElement>(
     onUpdateItemVisibility,
   ]);
 
-  const registerItem = React.useCallback(
-    (item: OverflowItemEntry) => {
-      overflowManager?.addItem(item);
-      item.element.setAttribute(DATA_OVERFLOW_ITEM, '');
-
-      return () => {
-        item.element.removeAttribute(DATA_OVERFLOWING);
-        item.element.removeAttribute(DATA_OVERFLOW_ITEM);
-        overflowManager?.removeItem(item.id);
-      };
-    },
-    [overflowManager],
-  );
-
-  const registerDivider = React.useCallback(
-    (divider: OverflowDividerEntry) => {
-      const el = divider.element;
-      overflowManager?.addDivider(divider);
-      el && el.setAttribute(DATA_OVERFLOW_DIVIDER, '');
-
-      return () => {
-        divider.groupId && overflowManager?.removeDivider(divider.groupId);
-        el.removeAttribute(DATA_OVERFLOW_DIVIDER);
-      };
-    },
-    [overflowManager],
-  );
-
-  const updateOverflow = React.useCallback(() => {
-    overflowManager?.update();
-  }, [overflowManager]);
-
-  const registerOverflowMenu = React.useCallback(
-    (el: HTMLElement) => {
-      overflowManager?.addOverflowMenu(el);
-      el.setAttribute(DATA_OVERFLOW_MENU, '');
-
-      return () => {
-        overflowManager?.removeOverflowMenu();
-        el.removeAttribute(DATA_OVERFLOW_MENU);
-      };
-    },
-    [overflowManager],
-  );
-
   return {
+    ...bindingsRef.current,
     containerRef,
-    registerItem,
-    updateOverflow,
-    registerOverflowMenu,
-    registerDivider,
   };
```

Comment:
I don't understand `bindingsRef.current`, it should be just:

```ts
return React.useMemo(() => ({
  ...createReactBindings(overflowManager),
  containerRef
}, [containerRef, overflowManager])
```

Reply:
The memo won't work, because the register functions need to be recreated. The register functions are callback ref  which need to be re-called when overflow options change. The easiest way to do this is to make sure they change reference

Reply:
So this works because when the overflow manager starts observing it triggers a render - because ResizeObserver is built to dispatch an event on initial observe to get the initial size.

I've updated the bindings to be stored in state anyway now so that from a React PoV it follows the render cycle correctly